dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/SessionInfo.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal;
+
+import java.time.Instant;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ *
+ */
+
+public class SessionInfo {
+    private final String sessionId;
+    private final Instant created;
+    private Instant lastAccessed;
+
+    private final Set<String> activeRequests = ConcurrentHashMap.newKeySet();
```

Comment:
Nothing is ever added to `activeRequests`. I'd expect to be adding to it when a tool call is starting and removing from it when the tool call ends. Finding a request to cancel should also use the session ID as part of the key. Please work with @ttmunyunguma who did the cancellation work.